### ISSUE TITLE
Can't change encoding on modified unsaved file.

### ISSUE THREAD

**Issue by Ben3eeE** [atom/encoding-selector#33]
## Description
If you modify a file with a file with a  non utf-8 encoding and then change the encoding (before saving) the encoding does not update but the status bar and encoding selector updates.

#### Steps to create a file with a non utf-8 encoding
1. Open atom in safe mode.
2. Create a new tab and change encoding. I used cp1252.
3. Add some text with special characters. I added `ääöäåöåöäåö`
4. Save the file as cp1252.txt.

#### Verify that encoding selector works normally
1. Open a special encoding file or cp1252.txt.
2. File looks garbled, this is expected.
3. Change encoding to cp1252 or relevant, file looks ok.
4. Back to utf-8 and garbled.

#### Issue I am describing
1. Open a special encoding file or cp1252.txt.
2. Make any modifications, I added a single `a` to 2nd empty line.
3. Change encoding to cp1252, `status-bar` and `encoding-selector` updates  but not the file.
4. You are forced to save with another encoding or ctrl-z all changes and change encoding.

### Expected
Atom to change encoding since the file was never saved or notify that changing encoding failed.

### Actual
Encoding never changed. Forced to lose changes or save with wrong encoding. No indication that a fail occurred.

**Comment by Dadibom** [comment]
For me, the file was ruined. I was not able to undo the encoding change. :(

**Comment by Ben3eeE** [comment]
Yeah, same for me. I mess up files all the time because of this bug. I have some files which have to be `cp-1252` and every time I edit them I forget about this 😀

**Comment by 50Wliu** [comment]
Did a bit of investigation on this.  Looks like:
1. encoding-selector calls `atom.workspace.getActiveTextEditor().setEncoding(id)`
2. TextEditor calls `TextBuffer.setEncoding(id)`
3. TextBuffer calls `file.setEncoding(id)`
4. Pathwatcher finally **assigns** the encoding, but doesn't update it until the file is read or saved.  This happens automatically when the file isn't modified because TextBuffer will reread the file once the encoding is changed.

Then it looks like we're still using the UTF-8 encodings to save to CP1252 because the modified contents don't get updated.  I'm not sure on this last part though.

If I'm correct, it looks like we might need to also include iconv-lite as a TextBuffer dependency.
